src/main/java/de/srendi/advancedperipherals/common/util/ServerWorker.java

```diff
@@ -5,23 +5,28 @@
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 import net.minecraftforge.fml.common.Mod;
 
-import java.util.ArrayDeque;
 import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
 
 @Mod.EventBusSubscriber(modid = AdvancedPeripherals.MOD_ID)
 public class ServerWorker {
 
-    private static final Queue<Runnable> callQueue = new ArrayDeque<>();
+    private static final Queue<Runnable> callQueue = new ConcurrentLinkedQueue<>();
 
     public static void add(final Runnable call) {
-        callQueue.add(call);
+        if (call != null) {
+            callQueue.add(call);
+        }
     }
 
     @SubscribeEvent
     public static void serverTick(TickEvent.ServerTickEvent event) {
         if (event.phase == TickEvent.Phase.END) {
-            while (!callQueue.isEmpty()) {
+            while (true) {
```

Comment:
Is there a reason why you switched from `while (!callQueue.isEmpty()) {` to `while (true) {`

Reply:
Because when `poll()` returns `null`, the queue is empty, just want take a less step of check